src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,221 @@
+import duke.tasks.Event;
+import duke.tasks.TaskType;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Class to store and manage Task objects.
+ */
+public class TaskList {
+
+    List<Task> taskList;
+
+    /**
+     * Create an empty TaskList.
+     */
+    public TaskList() {
+        taskList = new ArrayList<>();
+    }
+
+    /**
+     * Creates a TaskList from the given list of representations.
+     *
+     * @param stringList List of string representations
+     */
+    public TaskList(List<String> stringList) {
+        taskList = new ArrayList<>();
+        stringList.forEach(this::parseLine);
+    }
+
+    /**
+     * Returns the list of Tasks.
+     *
+     * @return List of Tasks
+     */
+    public List<Task> getList() {
+        return taskList;
+    }
+
+    /**
+     * Returns the size of the list.
+     *
+     * @return size of list
+     */
+    public int getSize() {
+        return taskList.size();
+    }
+
+    /**
+     * Creates a Task from the given representation and adds it to the list.
+     *
+     * @param line string repesentation of the Task
+     */
+    private void parseLine(String line) {
+        String[] lineSplit = line.split("\\|");
+        boolean done = lineSplit[2].equals("1");
+        if (lineSplit[0].equals("T")) {
+            addTask(lineSplit[1], TaskType.TODO, "", done);
+        } else if (lineSplit[0].equals("D")) {
+            addTask(lineSplit[1], TaskType.DEADLINE, lineSplit[3], done);
+        } else {
+            addTask(lineSplit[1], TaskType.EVENT, lineSplit[3], done);
+        }
+    }
+
+    /**
+     * Adds a Task to the list.
+     *
+     * @param task Task name
+     * @param type TaskType of the of Task to add
+     * @param ddl Task deadline
+     * @param done whether task has been marked done
+     */
+    private void addTask(String task, TaskType type, String ddl, boolean done) {
+        Task newTask;
+        if (type == TaskType.TODO) {
+            newTask = new Todo(task);
+        } else if (type == TaskType.DEADLINE) {
+            newTask = new Deadline(task, LocalDate.parse(ddl));
+        } else {
+            newTask = new Event(task, LocalDate.parse(ddl));
+        }
+
+        if (done) {
+            newTask.markDone();
+        }
+        taskList.add(newTask);
+    }
+
+    /**
+     * Add a Todo with the given name.
+     *
+     * @param task Todo name
+     * @return Task object added
+     */
+    public Task addTodo(String task) {
+        return addTask(new Todo(task));
+    }
+
+    /**
+     * Add a Deadline or Event with given attributes.
+     *
+     * @param taskAttr Task attributes
+     * @param isEvent whether Task is an Event
+     * @return Task object added
+     * @throws DukeException Duke-related exception due to erroneous inputs
+     */
+    public Task addDDLTask(String taskAttr, boolean isEvent) throws DukeException {
```

Comment:
Maybe we should change the method name to addDdlTask instead to adhere to the Coding Standard